entitlement/src/main/java/com/hevelian/identity/entitlement/evaluator/EntitlementEngineForPAPPolicy.java

```diff
@@ -0,0 +1,122 @@
+package com.hevelian.identity.entitlement.evaluator;
+
+import com.hevelian.identity.entitlement.ctx.RequestCtxFactory2;
+import com.hevelian.identity.entitlement.evaluator.finder.PAPFinderModule;
+import com.hevelian.identity.entitlement.model.pap.PAPPolicy;
+import org.w3c.dom.Element;
+import org.wso2.balana.PDP;
+import org.wso2.balana.PDPConfig;
+import org.wso2.balana.ParsingException;
+import org.wso2.balana.ctx.RequestCtxFactory;
+import org.wso2.balana.ctx.ResponseCtx;
+import org.wso2.balana.ctx.xacml3.RequestCtx;
+import org.wso2.balana.finder.AttributeFinder;
+import org.wso2.balana.finder.AttributeFinderModule;
+import org.wso2.balana.finder.PolicyFinder;
+import org.wso2.balana.finder.PolicyFinderModule;
+import org.wso2.balana.finder.impl.CurrentEnvModule;
+import org.wso2.balana.finder.impl.SelectorModule;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Evaluates policy by content or attributes.
+ */
+public class EntitlementEngineForPAPPolicy {
+  private final PDPConfig pdpConfig;
+  private final PDP pdp;
+
+  /**
+   * Constructs a new <code>PDP</code> object.
+   *
+   * @param papPolicy the {@link PAPPolicy} object.
+   */
+  public EntitlementEngineForPAPPolicy(PAPPolicy papPolicy) {
```

Comment:
This class is the same comparing to EntitlementEngine except the policy finder module. Why did you copy-paste all the sources?

Reply:
Created basic EntitlementEngine class for PAPEntitlementEngine and PDPEntitlementEngine.